src/EnergyPlus/OutputProcessor.cc

```diff
@@ -5091,311 +3593,317 @@ void UpdateDataandReport(EnergyPlusData &state, OutputProcessor::TimeStepType co
+                    }
+                } else {
+                    assert(false);
                 }
-
-                bool ReportNow = true;
-                if (rVar.SchedPtr > 0) ReportNow = (GetCurrentScheduleValue(state, rVar.SchedPtr) != 0.0); // SetReportNow(RVar%SchedPtr)
-                if (!ReportNow || !rVar.Report) {
-                    rVar.TSValue = 0.0;
-                }
-                //        IF (RVar%StoreType == AveragedVar) THEN
-                //          RVar%Value=RVar%Value+RVar%TSValue/NumOfTimeStepInHour
-                //        ELSE
-                rVar.Value += rVar.TSValue;
-                //        ENDIF
-
-                if (!ReportNow || !rVar.Report) continue;
-
-                if (rVar.frequency == ReportingFrequency::TimeStep) {
-                    if (TimePrint) {
-                        if (op->LHourP != state.dataGlobal->HourOfDay || std::abs(op->LStartMin - StartMinute) > 0.001 ||
-                            std::abs(op->LEndMin - op->TimeValue.at(thisTimeStepType).CurMinute) > 0.001) {
-                            int CurDayType = state.dataEnvrn->DayOfWeek;
-                            if (state.dataEnvrn->HolidayIndex > 0) {
-                                CurDayType = state.dataEnvrn->HolidayIndex;
-                            }
-                            WriteTimeStampFormatData(state,
-                                                     state.files.eso,
-                                                     ReportingFrequency::EachCall,
-                                                     op->TimeStepStampReportNbr,
-                                                     op->TimeStepStampReportChr,
-                                                     state.dataGlobal->DayOfSimChr,
-                                                     true,
-                                                     state.dataEnvrn->Month,
-                                                     state.dataEnvrn->DayOfMonth,
-                                                     state.dataGlobal->HourOfDay,
-                                                     op->TimeValue.at(thisTimeStepType).CurMinute,
-                                                     StartMinute,
-                                                     state.dataEnvrn->DSTIndicator,
-                                                     ScheduleManager::dayTypeNames[CurDayType]);
-                            op->LHourP = state.dataGlobal->HourOfDay;
-                            op->LStartMin = StartMinute;
-                            op->LEndMin = op->TimeValue.at(thisTimeStepType).CurMinute;
+            } // for (iMeter)
+        } // if (op->meterValues.capacity() > 0)
+
+        for (auto *var : op->outVars) {
+            if (var->timeStepType != TimeStepType::Zone && var->timeStepType != TimeStepType::System) continue;
+            if (var->varType != VariableType::Real) continue;
```

Comment:
This is nonsense.  I had to replicate this loop (and a few others) and process integer and real variables separately just to control the amounts of ordering diffs in ESO files so that I could debug.

Reply:
I assume this could be addressed further in a tiny PR which, while it may cause lots of diffs, would be a small bit of code to review and accept.

Reply:
Yes, it will cause a massive amount of ESO and MTR ordering diffs but the ultimate results will not change (FLW).